Lib/test/_test_multiprocessing.py

```diff
@@ -5849,6 +5850,24 @@ def test__all__(self):
         support.check__all__(self, multiprocessing, extra=multiprocessing.__all__,
                              not_exported=['SUBDEBUG', 'SUBWARNING'])
 
+    def test_spawn_sys_executable_none_allows_import(self):
+        # Regression test for a bug introduced in
+        # https://github.com/python/cpython/issues/90876 that caused an
+        # ImportError in multiprocessing when sys.executable was None.
+        # This can be true in embedded environments.
+        testfn = os_helper.TESTFN
+        self.addCleanup(os_helper.unlink, testfn)
+        with open(testfn, 'wb') as f:
+            f.write(b'''\
+import sys
+sys.executable = None
+assert "multiprocessing" not in sys.modules, "mp already imported!"
+import multiprocessing
+import multiprocessing.spawn  # This should not fail\n''')
+        rc, _, err = script_helper.assert_python_ok(testfn)
+        self.assertEqual(rc, 0)
+        self.assertEqual(err, b'')
+
```

Comment:
Actually I think this whole test can be a lot simpler, something along the lines of:
```suggestion
    def test_spawn_sys_executable_none_allows_import(self):
        # Regression test for a bug introduced in
        # https://github.com/python/cpython/issues/90876 that caused an
        # ImportError in multiprocessing when sys.executable was None.
        # This can be true in embedded environments.
        rc, _, err = script_helper.assert_python_ok('-c', '''if 1:
            import sys
            sys.executable = None
            assert "multiprocessing" not in sys.modules, "mp already imported!"
            import multiprocessing
            import multiprocessing.spawn  # This should not fail
        ''')
        self.assertEqual(rc, 0)
        self.assertEqual(err, b'')

```

Reply:
indeed, done. i also added a skip when not being run in the test_multiprocessing_spawn suite. _(with a common decorator for that - there are various other things in this suite that should ultimately be annotated as such or moved out to a non-templated-rerun-in-all-modes suite)_